classes/01class/exercises/c01-aws01/zarajoy/c01-aws01.md

```diff
@@ -0,0 +1,398 @@
+            "NetworkInterfaces": [
+                {
+                    "Attachment": {
+                        "AttachTime": "2020-06-16T06:43:58.000Z",
+                        "AttachmentId": "eni-attach-0f2eb3e32c4e8fde4",
+                        "DeleteOnTermination": true,
+                        "DeviceIndex": 0,
+                        "Status": "attaching"
+                    },
+                    "Description": "",
+                    "Groups": [
+                        {
+                            "GroupName": "SecureGroup",
+                            "GroupId": "sg-09c8bcfe909954f1a"
+                        }
+                    ],
+                    "Ipv6Addresses": [],
+                    "MacAddress": "06:87:b3:bd:c3:20",
+                    "NetworkInterfaceId": "eni-01e2e4d50fd097c6b",
+                    "OwnerId": "512742231244",
+                    "PrivateDnsName": "ip-172-31-45-4.ap-southeast-2.compute.internal",
+                    "PrivateIpAddress": "172.31.45.4",
+                    "PrivateIpAddresses": [
+                        {
+                            "Primary": true,
+                            "PrivateDnsName": "ip-172-31-45-4.ap-southeast-2.compute.internal",
+                            "PrivateIpAddress": "172.31.45.4"
+                        }
+                    ],
+                    "SourceDestCheck": true,
+                    "Status": "in-use",
+                    "SubnetId": "subnet-48638000",
+                    "VpcId": "vpc-ce545da9",
+                    "InterfaceType": "interface"
+                }
+            ],
+            "RootDeviceName": "/dev/xvda",
+            "RootDeviceType": "ebs",
+            "SecurityGroups": [
+                {
+                    "GroupName": "SecureGroup",
+                    "GroupId": "sg-09c8bcfe909954f1a"
+                }
+            ],
+            "SourceDestCheck": true,
+            "StateReason": {
+                "Code": "pending",
+                "Message": "pending"
+            },
+            "VirtualizationType": "hvm",
+            "CpuOptions": {
+                "CoreCount": 1,
+                "ThreadsPerCore": 1
+            },
+            "CapacityReservationSpecification": {
+                "CapacityReservationPreference": "open"
+            },
+            "MetadataOptions": {
+                "State": "pending",
+                "HttpTokens": "optional",
+                "HttpPutResponseHopLimit": 1,
+                "HttpEndpoint": "enabled"
+            }
+        }
+    ],
+    "OwnerId": "512742231244",
+    "ReservationId": "r-05d1863c7d002fa30"
+}
+- Commands to connect to the second EC2 instance:
+z@bacon:~$ eval 'ssh-agent'
+SSH_AUTH_SOCK=/tmp/ssh-22FZ7c7j6uMf/agent.203040; export SSH_AUTH_SOCK;
+SSH_AGENT_PID=203041; export SSH_AGENT_PID;
+echo Agent pid 203041;
+z@bacon:~$ aws ec2 describe-instances --query "Reservations[].Instances[].[InstanceId, PublicIpAddress]"
```

Comment:
the second instance doesn't have a public IP address attached.

Reply:
correct. only the one from the first instance does.
    [
        "i-08b574c6436ac7ad8",
        null
    ],